README.md

```diff
@@ -1 +1,197 @@
 # js-libp2p-circuit
+
+[![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://ipn.io)
+[![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs)
+[![Build Status](https://travis-ci.org/libp2p/js-libp2p-circuit.svg?style=flat-square)](https://travis-ci.org/libp2p/js-libp2p-circuit)
+[![Coverage Status](https://coveralls.io/repos/github/libp2p/js-libp2p-circuit/badge.svg?branch=master)](https://coveralls.io/github/libp2p/js-libp2p-circuit?branch=master)
+[![Dependency Status](https://david-dm.org/libp2p/js-libp2p-circuit.svg?style=flat-square)](https://david-dm.org/libp2p/js-libp2p-circuit)
+[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/feross/standard)
+![](https://img.shields.io/badge/npm-%3E%3D3.0.0-orange.svg?style=flat-square)
+![](https://img.shields.io/badge/Node.js-%3E%3D4.0.0-orange.svg?style=flat-square)
+
+![](https://raw.githubusercontent.com/libp2p/interface-connection/master/img/badge.png)
+![](https://raw.githubusercontent.com/libp2p/interface-transport/master/img/badge.png)
+
+> Node.js implementation of the Circuit module that libp2p uses, which implements the [interface-connection](https://github.com/libp2p/interface-connection) interface for dial/listen.
+
+`libp2p-circuit` implements the circuit-relay mechanism that allows nodes that don't speak the same protocol to communicate using a third _relay_ node.
+
+**Note:** This module uses [pull-streams](https://pull-stream.github.io) for all stream based interfaces.
+
+### Why?
+
+`circuit-relaying` uses additional nodes in order to transfer traffic between two otherwise unreachable nodes. This allows nodes that don't speak the same protocols or are running in limited environments, e.g. browsers and IoT devices, to communicate, which would otherwise be impossible given the fact that for example browsers don't have any socket support and as such cannot be directly dialed.
+
+The applicability of circuit-relaying is not limited to routing traffic between browser nodes, other uses include:
+ - routing traffic between private nets
+ - circumventing NAT layers
+ - route mangling for better privacy (matreshka/shallot dialing).
+
+ It's also possible to use it for clients that implement exotic transports such as  devices that only have bluetooth radios to be reachable over bluethoot enabled relays and become full p2p nodes.
```

Comment:
tiny typo: `bluethoot`

Reply:
that readme needs some serious work tho...